graphql-ballerina/tests/11_listener_configurations_test.bal

```diff
@@ -122,4 +135,90 @@ isolated function testQueryExceedingMaxDepth() returns error? {
     test:assertEquals(actualPayload, expectedPayload);
 }
 
+@test:Config {
+    groups: ["configs", "unit"]
+}
+isolated function testFragmentQueryExceedingMaxDepth() returns error? {
+    string document = string
+`query {
+    ...on Query {
+        people {
+            ... on Person {
+                address {
+                    city
+                }
+            }
+        }
+    }
+ }
+ query {
+    Person {
+        Address {
+            city
+        }
+    }
+ }
+ query {
+    Student {
+        name
+    }
+ }`;
+    string url = "http://localhost:9103/depthLimitService";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = {
+        errors: [
```

Comment:
We need to check the query names are correct with the positions. So we should use named operations here.

This should return another error, `This anonymous operation must be the only defined operation.`

A document cannot have multiple anonymous queries. If this does not return that error, shall we create an issue?

Reply:
```query_depth_validator``` execute before the ```validator_visitor```. Therefore that error won't hit.